### PR TITLE
Feature/part based camera

### PR DESCRIPTION
**Pull request by jvdwijk**
het eerste feessie

### CODE REVIEW

File: Assets/Scripts/Camera/Bounds/BoundList.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    [SerializeField]
+    private ZoomedOutState zoomedOutState;  
+
+    [SerializeField]
+    private LevelBoundsCalculator[] levelBoundsCalculators;
+
+    private void Start()
+    {
+        FindObjectOfType<SceneLevelHandler>().OnPlayerSetup += (janneke) => FindPlayerBounds();
```
Comment (CoenSteuns) [inline Assets/Scripts/Camera/Bounds/BoundList.cs]: Who is Janneke? David's competition?

File: Assets/Scripts/Camera/Bounds/LevelBoundsCalculator.cs [hunk 2]
```diff
@@ -28,4 +33,29 @@ public void UpdateBounds()
...
 
+    public void OnPlayerEntered()
+    {
+        checkBoundsRoutine = StartCoroutine(CheckBounds());
+    }
+
+    public bool IsPlayerInBounds()
+    {
+        if (playerTransform == null)
+            playerTransform = FindObjectOfType<PlayerJump>();
```
Comment (CoenSteuns) [inline Assets/Scripts/Camera/Bounds/LevelBoundsCalculator.cs:44]: FindObjectOfType is a little sad :(

File: Assets/Scripts/Camera/Bounds/BoundList.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    }
+
+    private void FindPlayerBounds()
+    {
+        LevelBoundsCalculator currentLevelBounds = null;
+        for (int i = 0; i < levelBoundsCalculators.Length; i++)
+        {
+            if(levelBoundsCalculators[i].isActiveAndEnabled)
+                currentLevelBounds = levelBoundsCalculators[i].IsPlayerInBounds() ? levelBoundsCalculators[i] : currentLevelBounds;
```
Comment (CoenSteuns) [inline Assets/Scripts/Camera/Bounds/BoundList.cs]:
```
if(levelBoundsCalculators[i].isActiveAndEnabled)
                currentLevelBounds = levelBoundsCalculators[i].IsPlayerInBounds() ? levelBoundsCalculators[i] : currentLevelBounds; 
```

Would be better as 

```
if(levelBoundsCalculators[i].isActiveAndEnabled && levelBoundsCalculators[i].IsPlayerInBounds())
                currentLevelBounds = levelBoundsCalculators[i]; 
```
Like this, you'd only have one if-statement

**Review by CoenSteuns — COMMENTED** [review]
:)

File: Assets/Scripts/Camera/Bounds/BoundList.cs (hunk 1 above)
Comment (jvdwijk) [reply Assets/Scripts/Camera/Bounds/BoundList.cs]: Janneke is lief beessie

File: Assets/Scripts/Camera/Bounds/LevelBoundsCalculator.cs (hunk 2 above)
Comment (jvdwijk) [reply Assets/Scripts/Camera/Bounds/LevelBoundsCalculator.cs:44]: zeker erg sad :(

**Pull request merged**